docs/concepts/what-is-a-workflow-execution.md

```diff
@@ -2,57 +2,122 @@
+description: A Temporal Workflow Execution is a durable, scalable, reliable, and reactive function execution. It is a Temporal Application’s main unit of execution.
 tags:
   - explanation
 ---
 
-A Workflow Execution is a Reentrant Process; that is, a resumable, recoverable, and reactive process:
+A Temporal Workflow Execution is a durable, reliable, and scalable function execution.
+It is a Temporal Application’s main unit of execution.
 
-- Resumable: Ability of a process to continue execution after execution was suspended on an await-able.
-- Recoverable: Ability of a process to continue execution after execution was suspended on a failure.
-- Reactive: Ability of a process to react to external events.
+- [How to spawn a Workflow Execution in Go](/docs/go/how-to-spawn-a-workflow-execution-in-go)
+- [How to spawn a Workflow Execution in Java](/docs/java/how-to-spawn-a-workflow-execution-in-java)
+
+Each Temporal Workflow Execution has exclusive access to its local state, executes concurrently to all other Workflow Executions, and communicates with other Workflow Executions and the environment via message passing.
+
+A Temporal Application can consist of millions to billions of Workflow Executions.
+Workflow Executions are lightweight components.
+A Workflow Execution consumes few compute resources; in fact, if a Workflow Execution is suspended, such as when it is in a waiting state, the Workflow Execution consumes no compute resources at all.
+
+**Durability**
+
+Durability is responsiveness over a period time.
+A Workflow Execution is durable because it executes a Temporal Workflow Definition, also called a Temporal Workflow Function, your application code, effectively once and to completion—whether your code executes for seconds or years.
+Durability is realized through the execution semantics of a Workflow Execution, which are carried out via Temporal's Event loop.
+
+**Reliability**
+
+Reliability is responsiveness in the presence of failure.
+A Workflow Execution is reliable, because it is fully recoverable after a failure.
+To be recoverable is the ability of a process to continue execution after the execution was suspended by a failure.
+Reliability is realized through the execution semantics of a Workflow Execution, which are carried out via Temporal's Event loop.
+
+**Scalability**
+
+Scalability is responsiveness in the presence of load.
+A Workflow Execution is scalable because the Temporal Platform is capable of supporting millions to billions of them concurrently.
+This is realized in part due to the design and nature of a [Temporal Cluster](/docs/concepts/what-is-a-temporal-cluster) and in part due to the design and nature of [Worker Processes](/docs/concepts/what-is-a-worker-process).
+
+### Commands
+
+A Workflow Execution does just two things:
+
+1. Issue [Commands](/docs/concepts/what-is-a-command)
+2. Wait on an Awaitable provided from the generation of Commands.
+
```

Comment:
Commands and awaitables are not concepts that are directly exposed to the user. Is it possible to rewrite this section that is more concrete? For example, that workflow is allowed to call only Temporal APIs for blocking operations.